crates/bevy_ecs/src/query/iter.rs

```diff
@@ -176,12 +176,15 @@ where
                 table,
             );
 
-            let table_row = archetype.entity_table_row(location.index);
             // SAFETY: set_archetype was called prior.
             // `location.index` is an archetype index row in range of the current archetype, because if it was not, the match above would have `continue`d
-            if F::filter_fetch(&mut self.filter, entity, table_row) {
+            if F::filter_fetch(&mut self.filter, entity, location.table_row as usize) {
```

Comment:
F::filter_fetch should always be taking at most a `u32` there, right? We should change its argument types.

Also see above about how new types would make this code clearer.

Reply:
This is a pretty big change and already done in #4878. I'll defer this to that PR.